### DISCUSSION TITLE
When I initiate a new request(defer is used in loader), will return a Promise with _tracked=true structure

### DISCUSSION THREAD

**Discussion by wmdmomo** [remix-run/remix#7855] [Q&A]
When I click the button, the last cached data will be printed out first, and then the new data will be returned, and in the new Promise structure, _tracked is encapsulated

![image](https://github.com/remix-run/remix/assets/43945660/f2e153bd-d1ff-41f5-9c39-13439861eab1)

``` typescript
export const loader = async () => {
  const isAuth = await getAuth();
  const welfare = getWelfare();
  return defer({
    isAuth,
    welfare,
  });
};

export default function Index() {
  const { isAuth, welfare } = useLoaderData<typeof loader>();
  console.log('state', isAuth, welfare);
  const { revalidate } = useRevalidator();
  return (
    <div>
      <button onClick={revalidate}>revalidate</button>
    </div>
  );
}
```

**Comment by kiliman** [comment]
If you print out `revalidator.state`, you'll see why this is happening. Remix is transitioning from `idle` => `loading` => `idle`. During the `loading` state, you see the current values before the new values are returned, and it transitions back to `idle`.

```ts
export default function Index() {
  const { isAuth, welfare } = useLoaderData<typeof loader>();
  const { state, revalidate } = useRevalidator();
  console.log({state, isAuth, welfare});
  return (
    <div>
      <button onClick={revalidate}>revalidate</button>
    </div>
  );
}
```

**Reply by wmdmomo** [reply]
@kiliman Thank you for your reply, but I still have a question, why is a Promise structure with _tracked=true returned after revalidate?

**Reply by wmdmomo** [reply]
@kiliman I noticed that the `<Awiat/>` component will check whether the passed promise is` _tracked=true.` Is the design of returning a Promise with _tracked=true after revalidation so that `<Suspense/>` will not render the fallback again?

**Reply by kiliman** [reply]
Not sure. Perhaps an implementation detail from Remix's `defer` support.
